src/frontend/gatsby-node.js

```diff
@@ -1,4 +1,40 @@
 const path = require(`path`);
+const fs = require('fs').promises;
+const fetch = require('node-fetch');
+
+exports.onPreInit = async function () {
+  const clientId = process.env.unsplashClientId || '';
+  const collectionId = process.env.collectionId || '9975402';
+
+  const frontEndPath = path.join('src', 'images', 'backgrounds');
+
+  if (clientId && collectionId) {
+    const response = await fetch(
+      `https://api.unsplash.com/collections/${collectionId}/photos/?client_id=${clientId}&per_page=100`,
+      {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    );
+    const data = await response.json();
+
+    for (const photo of data) {
+      const downloadUrl = photo.urls.raw;
+      const fileName = `${frontEndPath}/${photo.id}.jpg`;
+
+      try {
+        fs.accessSync(fileName, 400);
+      } catch (err) {
+        // We expect it not to exist on deploys.
+        await fetch(downloadUrl)
+          .then((res) => res.buffer())
+          .then((buffer) => fs.writeFile(`${frontEndPath}/${photo.id}.jpg`, buffer))
+          .catch((imgErr) => console.error(imgErr));
+      }
+    }
+  }
+};
```

Comment:
This still isn't right.  I don't have an API key to test this locally, but I think this should be:

```js
async function downloadPhoto(url, filename) {
  const res = await fetch(url);
  const buf = await res.buffer();
  await fs.writeFile(filename, buf);
}

async function shouldDownload(filename) {
  try {
    await fs.access(filename, fs.FS_OK);
    // File already exists, skip
    return false;
  } catch (err) {
    // No such file, we need it
    return true;
  }
}

exports.onPreInit = async function () {
  const clientId = process.env.unsplashClientId || '';
  const collectionId = process.env.collectionId || '9975402';

  const frontEndPath = path.join('src', 'images', 'backgrounds');

  if (!(clientId && collectionId)) {
    return;
  }

  // Get the list of photos from our Unsplash.com collection
  const res = await fetch(
    `https://api.unsplash.com/collections/${collectionId}/photos/?client_id=${clientId}&per_page=100`,
    {
      headers: {
        'Content-Type': 'application/json',
      },
    }
  );
  const photos = await res.json();

  // Process this list, since we only need to download photos we don't already have locally
  const downloads = photos
    .map((photo) => ({
      filename: path.join(frontEndPath, `${photo.id}.jpg`),
      url: photo.urls.raw,
    }))
    .filter(shouldDownload);

  // Download any photos we don't have
  await Promise.all(downloads.map(({ url, filename }) => downloadPhoto(url, filename)));
};
```

Reply:
That worked.... adding your change.

Reply:
Awesome!